drivers/timer/arcv2_timer0.c

```diff
@@ -512,76 +178,69 @@ int z_clock_driver_init(struct device *device)
 	return 0;
 }
 
-#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
-static int sys_clock_suspend(struct device *dev)
+void z_clock_set_timeout(s32_t ticks, bool idle)
 {
-	ARG_UNUSED(dev);
+	/* If the kernel allows us to miss tick announcements in idle,
+	 * then shut off
+	 * the counter. (Note: we can assume if idle==true that
+	 * interrupts are already disabled)
+	 */
+	if (IS_ENABLED(CONFIG_TICKLESS_IDLE) && idle && ticks == K_FOREVER) {
+		timer0_control_register_set(0);
+		timer0_count_register_set(0);
+		timer0_limit_register_set(0);
+		last_load = TIMER_STOPPED;
+		return;
+	}
 
-	saved_limit = timer0_limit_register_get();
-	saved_control = timer0_control_register_get();
+#if defined(CONFIG_TICKLESS_KERNEL) && !defined(CONFIG_QEMU_TICKLESS_WORKAROUND)
```

Comment:
You only need the QEMU_TICKLESS_WORKAROUND logic if you have to run in qemu without icount enabled (which tries to return host time from the cycle counter).  ARC doesn't have a qemu target and IIRC nsim is cycle-exact.  You just want to test for TICKLESS_KERNEL here.